pkg/artifacts/artifacts_storage.go

```diff
@@ -59,6 +59,22 @@ func InitializeArtifactStorage(pr *v1alpha1.PipelineRun, c kubernetes.Interface,
 	return NewArtifactBucketConfigFromConfigMap(configMap)
 }
 
+// CleanupArtifactStorage will delete the PipelineRun's PVC if it exists.
+func CleanupArtifactStorage(pr *v1alpha1.PipelineRun, c kubernetes.Interface, logger *zap.SugaredLogger) error {
+	configMap, err := c.CoreV1().ConfigMaps(system.GetNamespace()).Get(v1alpha1.BucketConfigName, metav1.GetOptions{})
+	shouldCreatePVC, err := needsPVC(configMap, err, logger)
+	if err != nil {
+		return err
+	}
+	if shouldCreatePVC {
+		err = deletePVC(pr, c)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
 func needsPVC(configMap *corev1.ConfigMap, err error, logger *zap.SugaredLogger) (bool, error) {
```

Comment:
id love to see some test cases for `needsPVC` too if possible :)

Reply:
Reasonable!

Reply:
Actually, again, this is covered by `TestInitializeArtifactStorageWithConfigMap` and `TestInitializeArtifactStorageWithoutConfigMap`, I think? Does it really make sense to break that out into additional duplicative unit tests?

Reply:
Done.